source/parser/x86_64/type_checker.hpp

```diff
@@ -0,0 +1,38 @@
+// Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+// Spack Project Developers. See the top-level COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+#include <iostream>
+#include <optional>
+#include <regex>
+#include <stack>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "Function.h"
+#include "Symtab.h"
+#include "Type.h"
+#include "smeagle/parameter.h"
+
+namespace smeagle::x86_64 {
+
+  namespace st = Dyninst::SymtabAPI;
+
+  // Return true if the dataClass is a pointer type
+  static bool is_pointer(st::dataClass dc) { return dc == st::dataPointer; }
```

Comment:
If we were being super careful, we'd change all of these functions from `static` to `inline` to avoid ODR violations.

Reply:
okay this is done!